src/lib/scheduler/abstract_task.cpp

```diff
@@ -17,6 +17,15 @@ namespace hyrise {
 
 AbstractTask::AbstractTask(SchedulePriority priority, bool stealable) : _priority{priority}, _stealable{stealable} {}
 
+AbstractTask::~AbstractTask() {
+  // The scheduler takes task lists where each task might store shared_pointers to other tasks of the list (in form of
```

Comment:
This was the issue on macOS. I don't know why it cannot be debugged and why it leads to bus errors (vs. working fine on Linux), but the issue makes somewhat sense.

While the destructor apparently fixes the issue for now, I don't really like it. I was thinking about successors being stored differently. Either as references or also weak_ptrs. But I am not really sure what makes sense here.